src/components/PriceICP.js

```diff
@@ -1,36 +1,37 @@
-const _showListingPrice = n => {
+import {truncateNumber} from '../truncation';
+import {cssToReactStyleObject, toniqFontStyles, Icp16Icon, toniqColors} from '@toniq-labs/design-system';
+import {ToniqIcon} from '@toniq-labs/design-system/dist/esm/elements/react-components';
+
+function getIcpPrice(n) {
     n = Number(n) / 100000000;
     return n.toFixed(8).replace(/0{1,6}$/, '');
 };
-const numberWithCommas = x => {
+function numberWithCommas(x) {
     var parts = x.toString().split('.');
     parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',');
     return parts.join('.');
 };
 
-export function icpToString(price /* BigInt */, clean, volume) {
-  const initialPrice = clean
-    ? numberWithCommas(price)
-    : numberWithCommas(_showListingPrice(price));
-  const formattedPrice = volume
-    ? Number(initialPrice.replace(',', '')) >= 10000
-        ? (Number(initialPrice.replace(',', '')) / 1000).toFixed(1) + 'k'
-        : initialPrice
-    : initialPrice;
+export function icpToString(priceE8s /* BigInt */, convertToIcp, truncate) {
+  const initialPrice = convertToIcp
+    ? getIcpPrice(priceE8s)
+    : priceE8s;
   
-  return formattedPrice;
+  if (truncate) {
+    return truncateNumber(initialPrice);
+  } else {
+    return numberWithCommas(initialPrice);
+  }
 }
 
 export default function PriceICP(props) {
     return (
-        <>
-            <img
-                style={{verticalAlign: 'top'}}
-                src={'/currencies/icp.png'}
-                height={props.size ?? 18}
-                width={props.size ?? 18}
-            />{' '}
-            {icpToString(props.price, props.clean, props.volume)}
-        </>
+        <span style={{display: 'inline-flex', alignItems: 'center', gap: '4px'}}>
+        
+            <ToniqIcon icon={Icp16Icon}/>
```

Comment:
I adjusted this icon to use smaller image data but didn't change the image data dimensions. Fix in the design system repo.

Reply:
done.